Sodium/ExitCode.swift

```diff
@@ -0,0 +1,15 @@
+enum ExitCode {
```

Comment:
This is never used as an actual exit code, so I think the name might be slightly confusing. I suggest to rename it to status something.

In addition, each function might behave differently, e.g., `sodium_init()` can return `1`, which indicates that the library has already been initialized, but is not a failure technically. Therefore, I think a generic approach is more confusing than helpful. For syntactic sugaring we could use `noErr` or define a `SUCCESS` constant.

Reply:
The idea behind the name was to make implicit the general use-case that the integer `0` would correspond to the success state.

I take your point about it not being so easily useful when functions aren't as simple as `0` for success, and failure otherwise though.

We could perhaps add more states, but I'm now questioning the value add altogether. When this was written, at the use site there were lots of nested closures surrounding the function due to explicit pointer use (and so IMO this helped highlight some of the intent upfront because the actual function call could be something like five layers of nesting deep). This isn't really a concern any more, and so perhaps this is no longer justified since the function call is far more readable in the first place.